hack/generator/pkg/astmodel/storage_conversion_endpoint.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+import (
+	"github.com/gobuffalo/flect"
+	"strconv"
+	"strings"
+)
+
+// StorageConversionEndpoint represents either a source or a destination field for a storage conversion
+type StorageConversionEndpoint struct {
+	// theType is the Type of the value accessible via this endpoint
+	theType Type
+	// name is the name of the underlying property, used to generate useful local identifiers
+	name string
+	// knownLocals is a shared map of locals that have already been created within a given function, to prevent duplicates
+	knownLocals *KnownLocalsSet
+}
+
+func NewStorageConversionEndpoint(
+	theType Type,
+	name string,
+	knownLocals *KnownLocalsSet) *StorageConversionEndpoint {
+	return &StorageConversionEndpoint{
+		theType:     theType,
+		name:        strings.ToLower(name),
+		knownLocals: knownLocals,
+	}
+}
+
+// Expr returns a clone of the expression for this endpoint
```

Comment:
minor: Says `Expr` but function is called `Type`.

Also the comment doesn't seem to match the method